src/utils/dom.ts

```diff
@@ -215,3 +215,12 @@ export function setRequestedIcon(
     return iconObject[matchedValue];
   }
 }
+
+export function intersects(rect1: DOMRect, rect2: DOMRect): boolean {
```

Comment:
👏

Reply:
this is awesome!  Now it can be used for more than just the x direction.